packages/core/chrome/core-chrome-browser-internal/src/project_navigation/project_navigation_service.ts

```diff
@@ -48,11 +61,18 @@ export class ProjectNavigationService {
         return this.projectHome$.asObservable();
       },
       setProjectNavigation: (projectNavigation: ChromeProjectNavigation) => {
-        this.projectNavigation$.next(projectNavigation);
+        const { navigationTreeFlattened, ...rest } = projectNavigation;
+        this.projectNavigation$.next(rest);
```

Comment:
I don't know if there is a real bug here currently, but I think need to be extra careful here because `this.projectNavigation$.next(rest);` is sync and all subscribers would receive an update sync. if some of the subscribes use `this.projectNavigationNavTreeFlattened` or `activeNodes`, then they would see the previous not yet updates state

Reply:
Makes sense, great catch 👍 I'll look into it

Reply:
With the latest changes this should be OK. There are 2 Observable that are updated in the same tick and should be in sync. 👍